wsmaster/che-core-api-workspace-activity/src/main/java/org/eclipse/che/api/workspace/activity/WorkspaceActivityDao.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.workspace.activity;
+
+import java.util.List;
+import org.eclipse.che.api.core.ServerException;
+
+/**
+ * Data access object for workspaces expiration times.
+ *
+ * @author Max Shaposhnik (mshaposh@redhat.com)
+ */
+public interface WorkspaceActivityDao {
+
+  /**
+   * Sets workspace expiration time. Any running workspace must prolongate expiration time
+   * periodically, otherwise it will be stopped by passing that time.
+   *
+   * @param expiration WorkspaceExpiration object
+   * @throws ServerException when operation failed
+   */
+  void setExpiration(WorkspaceExpiration expiration) throws ServerException;
+
+  /**
+   * Removes workspace expiration time on it's stop.
```

Comment:
Looks like a typo `it's` -> `its` ?
BTW DAO knows nothing about starting and stopping of a workspace.